src/aliceVision/keyframe/KeyframeSelector.cpp

```diff
@@ -25,452 +25,711 @@ namespace keyframe {
+                    break;
+                }
+                motionAcc = 0.0;
+
+                for (std::size_t i = 1; i < sequenceSize - 1; i++) {
+                    motionAcc += _flowScores.at(i);
+                    if (motionAcc >= step) {
+                        newLimits.push_back(i);
+                        motionAcc = 0.0;
+                    }
+                }
+                newLimits.push_back(sequenceSize - 1);
+            }
+
+            if (sampleRegularly) {
+                // Sample regularly the whole sequence to get minOutFrames subsequences
+                newLimits.clear();
+                newLimits.push_back(0);
+                std::size_t stepSize = (sequenceSize / _minOutFrames) + 1;
+
+                for (std::size_t i = 1; i < sequenceSize - 1; i += stepSize)
+                    newLimits.push_back(i);
+                newLimits.push_back(sequenceSize - 1);
+            }
+        } else {
+            // Too many frames, increase the motion step
+            while (newLimits.size() - 1 > _maxOutFrames) {
+                newLimits.clear();
+                newLimits.push_back(0);
+                step = step + displacementDiff;
+                motionAcc = 0.0;
+
+                for (std::size_t i = 1; i < sequenceSize - 1; i++) {
+                    motionAcc += _flowScores.at(i);
+                    if (motionAcc >= step) {
+                        newLimits.push_back(i);
+                        motionAcc = 0.0;
+                    }
+                }
+                newLimits.push_back(sequenceSize - 1);
+            }
+        }
 
-    {
-      if(frameSelected)
-      {
-        ALICEVISION_LOG_INFO(" > selected" << std::endl);
-        frameData.selected = true;
-        if(_hasSharpnessSelection)
-          frameData.computeAvgSharpness();
-      }
-      else
-      {
-        ALICEVISION_LOG_INFO(" > skipped" << std::endl);
-        frameData.mediasData.clear(); // remove unselected mediasData
-      }
+        subsequenceLimits.clear();
+        subsequenceLimits = newLimits;
     }
 
-    // selection process
-    if(currentFrameStep >= _maxFrameStep)
-    {
-      currentFrameStep = _minFrameStep;
-      bool hasKeyframe = false;
-      std::size_t keyframeIndex = 0;
-      float maxSharpness = 0;
-      float minDistScore = std::numeric_limits<float>::max();
-
-      // find the best selected frame
-      if(_hasSharpnessSelection)
-      {
-        // find the sharpest selected frame
-        for(std::size_t index = frameIndex - (frameStep - 1); index <= frameIndex; ++index)
-        {
-          if(_framesData[index].selected && (_framesData[index].avgSharpness > maxSharpness))
-          {
-            hasKeyframe = true;
-            keyframeIndex = index;
-            maxSharpness = _framesData[index].avgSharpness;
-          }
+    // Step 3: for each subsequence, find the keyframe
+    for (std::size_t i = 1; i < subsequenceLimits.size(); i++) {
```

Comment:
`++i` instead of `i++` (same in several other places below)

Reply:
Done.